src/main/java/no/unit/nva/doi/transformer/model/internal/external/DataciteTypes.java

```diff
@@ -1,44 +1,85 @@
 package no.unit.nva.doi.transformer.model.internal.external;
 
+import com.fasterxml.jackson.annotation.JsonProperty;
+import no.unit.nva.doi.transformer.utils.BibTexType;
+import no.unit.nva.doi.transformer.utils.CiteProcType;
+import no.unit.nva.doi.transformer.utils.RisType;
+import no.unit.nva.doi.transformer.utils.SchemaOrgType;
+
 public class DataciteTypes {
 
-    private String ris;
-    private String bibtex;
-    private String citeproc;
-    private String schemaOrg;
+    private RisType ris;
+    private BibTexType bibtex;
+    private CiteProcType citeproc;
+    private SchemaOrgType schemaOrg;
     private String resourceType;
     private String resourceTypeGeneral;
 
-    public String getRis() {
+    /**
+     * Constructor for Datacite types structures.
+     *
+     * @param ris the RIS type.
+     * @param bibtex the BibTeX type.
+     * @param citeproc the CiteProc type.
+     * @param schemaOrg the Schema.org type.
+     * @param resourceType the Datacite resource type.
+     * @param resourceTypeGeneral the Datacite resource type general.
+     */
+    public DataciteTypes(
```

Comment:
I think you have forgotten the `@JsonCreator` tag
```suggestion    
@JsonCreator
public DataciteTypes(
```